### PR TITLE
[@testing-library/react_v10.x.x] Add missing find* queries

### PR DESCRIPTION
**Pull request by samtsai**
Clean up a few other things to reflect more closely to the public Type Definitions

- Links to documentation: https://testing-library.com/docs/dom-testing-library/api-queries#findby
- Link to GitHub or NPM: https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/testing-library__dom
- Type of contribution: fix | refactor

### CODE REVIEW

**Comment by samtsai** [comment]
I'm honestly not sure why `test_react_v8.x.x` tests fail.  AFAIK I've only changed `react_v10.x.x`

**Comment by samtsai** [comment]
cc @pascalduez

**Comment by AndrewSouthpaw** [comment]
Our CI doesn't distinguish older versions unfortunately so if they're broken then they'll keep triggering build failures.

**Comment by pascalduez** [comment]
I would have preferred having enough time to review @AndrewSouthpaw.  

What worries me is that every contributor to those definitions feels the needs to "refactor", which leads to stubble, hard to track regressions and changes since we don't have such a great test coverage. 
Also making improvements hard to track over time, because everything constantly change.

On the other hand, trying to be as close as possible to the TS definitions is probably a good thing.

Thanks @samtsai

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js [hunk 1]
```diff
@@ -28,11 +35,33 @@ declare module '@testing-library/react' {
...
-  declare type GetByText = (
+  declare type QueryByBoundAttribute = (
     text: TextMatch,
-    options?: SelectorMatchOptions
+    options?: TextMatchOptions
+  ) => ?HTMLElement;
+
+  declare type AllByBoundAttribute = (
+    text: TextMatch,
+    options?: TextMatchOptions
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:45]: Missing `id` property.

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js [hunk 2]
```diff
@@ -28,11 +35,33 @@ declare module '@testing-library/react' {
...
+  ) => Array<HTMLElement>;
+
+  declare type FindAllByBoundAttribute = (
+    text: TextMatch,
+    options?: TextMatchOptions,
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:50]: Missing `id` property.

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js [hunk 3]
```diff
@@ -28,11 +35,33 @@ declare module '@testing-library/react' {
...
+    waitForElementOptions?: WaitForElementOptions
+  ) => Promise<HTMLElement[]>;
+
+  declare type GetByBoundAttribute = (
+    text: TextMatch,
+    options?: TextMatchOptions
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:56]: Missing `id` property.

File: definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js [hunk 4]
```diff
@@ -28,11 +35,33 @@ declare module '@testing-library/react' {
...
   ) => HTMLElement;
 
+  declare type FindByBoundAttribute = (
+    text: TextMatch,
+    options?: TextMatchOptions,
```
Comment (pascalduez) [inline definitions/npm/@testing-library/react_v10.x.x/flow_v0.104.x-/react_v10.x.x.js:61]: Missing `id` property.

**Comment by silvenon** [comment]
Thanks @samtsai! ❤️ I'm using this libdef in a team project, and I'm very happy to see that this is moving towards TS definitions, it will be much easier to contribute that way.

**Comment by samtsai** [comment]
Thanks @AndrewSouthpaw .

Yea @pascalduez I agree, I didn't want to unnecessarily refactor on existing but I was trying to base the new changes on the existing Typescript definitions.

**Pull request merged**